vapi/library/example_test.go

```diff
@@ -67,3 +72,403 @@ func ExampleManager_CreateLibrary() {
+			LibraryID: pubLib.ID,
+		})
+		if err != nil {
+			return err
+		}
+
+		pubItem, err := m.GetLibraryItem(ctx, pubItemID)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("  created library item", pubItem.Name)
+
+		uploadSessionID, err := m.CreateLibraryItemUpdateSession(
+			ctx,
+			library.Session{
+				LibraryItemID: pubItemID,
+			})
+
+		uploadFn := func(path string) error {
+			f, err := os.Open(filepath.Clean(path))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+
+			fi, err := f.Stat()
+			if err != nil {
+				return err
+			}
+
+			info := library.UpdateFile{
+				Name:       filepath.Base(path),
+				SourceType: "PUSH",
+				Size:       fi.Size(),
+			}
+
+			update, err := m.AddLibraryItemFile(ctx, uploadSessionID, info)
+			if err != nil {
+				return err
+			}
+
+			u, err := url.Parse(update.UploadEndpoint.URI)
+			if err != nil {
+				return err
+			}
+
+			p := soap.DefaultUpload
+			p.ContentLength = info.Size
+
+			return m.Client.Upload(ctx, f, u, &p)
+		}
+
+		if err := uploadFn("./testdata/ttylinux-pc_i486-16.1.ova"); err != nil {
+			return err
+		}
+
+		if err := m.CompleteLibraryItemUpdateSession(
+			ctx, uploadSessionID); err != nil {
+
+			return err
+		}
+
+		pubItemStor, err := m.ListLibraryItemStorage(ctx, pubItemID)
+		if err != nil {
+			return err
+		}
+
+		for i := range pubItemStor {
+			is := pubItemStor[i]
+			fmt.Printf(
+				"    uploaded library item file %s, cached=%v, size=%d\n",
+				is.Name, is.Cached, is.Size)
+		}
+
+		// Create a subscribed library that points to the one above.
+		subLibID, err := m.CreateLibrary(ctx, library.Library{
+			Name: "my-sub-lib",
+			Type: "SUBSCRIBED",
+			Storage: []library.StorageBacking{
+				{
+					DatastoreID: ds.Reference().Value,
+					Type:        "DATASTORE",
+				},
+			},
+			Subscription: &library.Subscription{
+				SubscriptionURL: pubLib.Publication.PublishURL,
+				OnDemand:        types.New(true),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		subLib, err := m.GetLibraryByID(ctx, subLibID)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("created library", subLib.Name)
+
+		subItemIDs, err := m.ListLibraryItems(ctx, subLibID)
+		if err != nil {
+			return err
+		}
+
+		for i := range subItemIDs {
+			subItemID := subItemIDs[i]
+
+			subItem, err := m.GetLibraryItem(ctx, subItemID)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println("  got subscribed library item", subItem.Name)
+
+			// List the item's storage prior to being synced.
+			subItemStor, err := m.ListLibraryItemStorage(ctx, subItemID)
+			if err != nil {
+				return err
+			}
+			for i := range subItemStor {
+				is := subItemStor[i]
+				fmt.Printf(
+					"    library item file %s, cached=%v, size=%d\n",
+					is.Name, is.Cached, is.Size)
+			}
+
+			// Synchronize the item.
+			if err := m.SyncLibraryItem(ctx, subItem, true); err != nil {
+				return err
+			}
+
+			fmt.Println("  sync'd subscribed library item", subItem.Name)
```

Comment:
nit: maybe print `force sync'd` to indicate the force param on the CL API is set to be true?

Reply:
Do you mind if I handle this in a follow-up so I don't lose the approvals?

Reply:
Nevermind, I had to re-push anyway due to an issue Is saw.